File: Kotlin/lambda - closure/4/main.kt
/**
 * Created by fay on 2017/5/25.
 */

fun main(args: Array<String>) {
    say { content: String ->
        println(content)
        "Hello Kotlin!"
    }
}

fun say(reply: (string: String) -> String) {
    println(reply("Hello World!"))
}

File: Scala/lambda - closure/4/main.scala
/**
  * Created by fay on 2017/6/15.
  */

object main {
  def main(args: Array[String]): Unit = {
    say(end => {
      println("Hello Scala!")
      end("^_^")
    })
  }

  def say(reply: ((String) => Unit) => Unit) = {
    println("Hello World!")
    reply { content =>
      println(content)
    }
  }
}
